### PR TITLE
[JFT-04] Fix search component for Home, and Favorite view

### PR DESCRIPTION
**Pull request by julianfigueroaGL**
The function filterBySearchTerm was added to Home and All List Favorite components.

JFT-04 - Search 01
JFT-04 - Search 02

### CODE REVIEW

File: src/views/list-all-favorites/list-all-favorites.js [hunk 1]
```diff
@@ -10,12 +11,20 @@ export default {
...
       detailPath: DETAIL_PATH,
+      favoriteListFiltered: [],
     };
   },
   components: {
     ListPodcast,
   },
+  methods: {
+    async filterFavorites(term) {
+      const termSanitized = term.toLowerCase().trim();
```
Comment (alejandrobenjumeaGL) [inline src/views/list-all-favorites/list-all-favorites.js]: please move this logic inside 'filterBySearchTerm' function

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

File: src/views/home/home.js [hunk 2]
```diff
@@ -29,26 +32,12 @@ export default {
...
-        this.favorites = [...this.initialFavorites];
-        return;
-      }
-
-      this.favorites = this.initialFavorites.filter((favorite) => {
-        const track = favorite.trackName.toLowerCase();
-        return track.includes(term.toLowerCase());
-      });
+    async filterFavorites(term) {
+      const termSanitized = term.toLowerCase().trim();
```
Comment (josepenagosGL) [inline src/views/home/home.js]: We already have a function for this into our utils (trim) Either move it to that function or create a new one to wrap up lowercase will be my suggestion

File: src/views/home/home.js [hunk 3]
```diff
@@ -79,12 +68,13 @@ export default {
     ListPodcast,
     recentPodcast,
   },
-  async created() {
+  async mounted() {
```
Comment (josepenagosGL) [inline src/views/home/home.js:69]: Just to clarify, why are we using mounted instead ?

**Review by josepenagosGL — CHANGES_REQUESTED** [review]

File: src/views/home/home.js (hunk 3 above)
Comment (julianfigueroaGL) [reply src/views/home/home.js:69]: Created will not wait for all of the async operations to complete before moving on to the next stage when making API call.

File: src/views/home/home.js (hunk 2 above)
Comment (julianfigueroaGL) [reply src/views/home/home.js]: Added to filterBySearchTerm function

File: src/views/list-all-favorites/list-all-favorites.js (hunk 1 above)
Comment (julianfigueroaGL) [reply src/views/list-all-favorites/list-all-favorites.js]: done

**Review by josepenagosGL — APPROVED** [review]

**Review by alejandrobenjumeaGL — APPROVED** [review]

**Pull request merged**